core/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/deltaMerge.scala

```diff
@@ -424,7 +424,12 @@ object DeltaMergeInto {
         // Note: This will throw error only on unresolved attribute issues,
         // not other resolution errors like mismatched data types.
         val cols = "columns " + plan.children.flatMap(_.output).map(_.sql).mkString(", ")
-        a.failAnalysis(msg = s"cannot resolve ${a.sql} in $mergeClauseType given $cols")
+        // todo: added a new Delta error for this to avoid rewriting tests, but existing
```

Comment:
Will there be a follow up PR for this? If so, please clarify what need to be done. This is a little vague for a TODO
```suggestion
        // TODO: added a new Delta error for this to avoid rewriting tests, but existing
```

Reply:
Ah just a note to myself / reviewers on the other option. I think I prefer using the new Delta error I will remove this comment